addons/mail/models/mail_notification.py

```diff
@@ -82,3 +82,34 @@ def _gc_notifications(self, max_age_days=180):
             ('notification_status', 'in', ('sent', 'canceled'))
         ]
         return self.search(domain).unlink()
+
+    @api.model
+    def _types_for_web_client(self):
```

Comment:
I don't really like the naming and _status_for_web_client does not seem to be used. I think we should not support exotic statuses for failed notification: bounce, exception and canceled should be sufficient (allowing to remove one method).

Concerning _types_for_web_client: shouldn't we exclude inbox instead of adding them one by one for each new selection key ? Maybe it lessens ease of override for external people but if it works for our standard use cases ... less methods, less code, less tests :D .

Reply:
Ok to remove `_status_for_web_client`, I originally needed it for snaimail but then I decided to use "exception" as you suggest.

`_types_for_web_client` can indeed be replaced by just excluding "inbox" in the main method, I asked myself the same question. I went for the separate method returning a specific list to give more rooms to customization and to make it less prone to side effects, but I don't mind to change if you prefer less code here.